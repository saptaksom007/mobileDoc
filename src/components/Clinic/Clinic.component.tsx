

import React, { PureComponent } from 'react'
import i18n from 'ex-react-native-i18n'

import { Bold, Medium } from 'components/CustomText/CustomText.component'
import PhoneNumber from 'components/PhoneNumber/PhoneNumber.component'

import styles from './Clinic.styles'

interface Props {
  name: string,
  contact: string,
  street: string,
  location: string
}

class Clinic extends PureComponent<Props> {
  render() {
    const { name, contact, street, location } = this.props
    return (
      <Medium style={styles.container}>
        <Bold>{i18n.t('clinic.name')}:</Bold> {name}
        {'\n'}
        <Bold>{i18n.t('clinic.address')}:</Bold> {street}
        {', '}
        {location}
        {'\n'}
        <Bold>{i18n.t('clinic.phone')}:</Bold> <PhoneNumber value={contact} />
        {'\n'}
      </Medium>
    )
  }
}

export default Clinic
