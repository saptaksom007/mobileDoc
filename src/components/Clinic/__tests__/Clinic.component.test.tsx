import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Clinic from '../Clinic.component'

jest.mock('components/PhoneNumber/PhoneNumber.component', () => 'PhoneNumber')

it('<Clinic/> render correctly', () => {
  const tree = renderer
    .create(
      <Clinic name="name" contact="0303003" location="wwww" street="asd" />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
