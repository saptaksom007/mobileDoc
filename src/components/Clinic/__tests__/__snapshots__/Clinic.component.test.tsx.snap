// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clinic/> render correctly 1`] = `
<Text
  allowFontScaling={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "TitilliumWeb-Regular",
        "fontSize": 17.5,
        "lineHeight": 22,
        "overflow": "visible",
      },
      Object {
        "color": "rgba(27, 106, 158, 0.85)",
        "fontSize": 16.875,
        "lineHeight": 20,
      },
      Object {
        "backgroundColor": "#fff",
        "paddingVertical": 20,
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 17.5,
          "lineHeight": 22,
          "overflow": "visible",
        },
        Object {
          "color": "rgba(27, 106, 158, 0.85)",
          "fontFamily": "TitilliumWeb-SemiBold",
        },
        undefined,
      ]
    }
  >
    clinic.name
    :
  </Text>
   
  name
  

  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 17.5,
          "lineHeight": 22,
          "overflow": "visible",
        },
        Object {
          "color": "rgba(27, 106, 158, 0.85)",
          "fontFamily": "TitilliumWeb-SemiBold",
        },
        undefined,
      ]
    }
  >
    clinic.address
    :
  </Text>
   
  asd
  , 
  wwww
  

  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 17.5,
          "lineHeight": 22,
          "overflow": "visible",
        },
        Object {
          "color": "rgba(27, 106, 158, 0.85)",
          "fontFamily": "TitilliumWeb-SemiBold",
        },
        undefined,
      ]
    }
  >
    clinic.phone
    :
  </Text>
   
  <PhoneNumber
    value="0303003"
  />
  

</Text>
`;
